service/src/main/java/bio/terra/workspace/app/controller/ControlledAwsResourceApiController.java

```diff
@@ -461,4 +461,56 @@ public ResponseEntity<ApiCreateControlledAwsSagemakerNotebookResult> createAwsSa
         getAwsSagemakerNotebookCreateResult(jobId);
     return new ResponseEntity<>(result, getAsyncResponseCode(result.getJobReport()));
   }
+
+  @Traced
+  @Override
+  public ResponseEntity<ApiAwsSagemakerNotebookResource> getAwsSagemakerNotebook(
+      UUID workspaceUuid, UUID resourceUuid) {
+    AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
+    ControlledAwsSagemakerNotebookResource resource =
+        controlledResourceMetadataManager
+            .validateControlledResourceAndAction(
+                userRequest,
+                workspaceUuid,
+                resourceUuid,
+                SamConstants.SamControlledResourceActions.READ_ACTION)
+            .castByEnum(WsmResourceType.CONTROLLED_AWS_SAGEMAKER_NOTEBOOK);
+    return new ResponseEntity<>(resource.toApiResource(), HttpStatus.OK);
+  }
+
+  @Traced
+  @Override
+  public ResponseEntity<ApiDeleteControlledAwsResourceResult> deleteAwsSagemakerNotebook(
+      UUID workspaceUuid,
+      UUID resourceUuid,
+      @Valid ApiDeleteControlledAwsResourceRequestBody body) {
+    return deleteAwsResource(workspaceUuid, resourceUuid, body);
```

Comment:
As a sanity check, it might be worth passing the expected resource type into `deleteAwsResource` and validating there. Otherwise it's possible to e.g. call `deleteAwsSagemakerNotebook` with a bucket's resource ID, which would be a bit confusing.